mitodl/micromasters
# Added Elasticsearch index

#### What are the relevant tickets?
Fixes #699 
Fixes #670 

#### What's this PR do?
Sets up Elasticsearch indexing to happen:
 - when the `recreate_index` management command is run.
 - when a `Profile`, `Education`, or `Employment` are updated or created. In this case a celery task is executed

#### Where should the reviewer start?
search/api.py

#### How should this be manually tested?
There are changes to `docker-compose.yml` so you'll need to restart your containers, no rebuilding should be necessary though.

First, run `docker-compose run web ./manage.py recreate_index`. Then run this command: `docker-compose run web curl http://elastic:9200/micromasters/_search | python -m json.tool`. Verify that you see profile data in the search results.

Go into the users page and make an update to the profile. Run the curl command again and verify that you see the updated value in the profile.

#### Background context
From now on, people creating new instances of this app will need to create an Elasticsearch index and mapping via management command: `./manage.py recreate_index`. This command may also be run at any time to delete the existing index and reindex all relevant data (this may take a while).
